src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -4190,18 +4135,29 @@ rmdir(const char *path)
 		goto out_org;
 	atomic_fetch_add_relaxed(&num_rmdir, 1);
 
-	rc = dfs_remove(dfs_mt->dfs, parent, item_name, false, NULL);
+	rc = dfs_remove(dfs_mt->dfs, drec2obj(parent), item_name, false, NULL);
 	if (rc)
 		D_GOTO(out_err, rc);
 
+	if (parent != NULL) {
+		rc = drec_del(dfs_mt->dcache, full_path, parent);
+		if (rc != -DER_SUCCESS && rc != -DER_NONEXIST)
+			DL_ERROR(rc, "DAOS directory cache cleanup failed");
+	}
+
+	drec_decref(dfs_mt->dcache, parent);
```

Comment:
Should this line, "drec_decref(dfs_mt->dcache, parent);", be removed? "drec_del(dfs_mt->dcache, full_path, parent);" was executed already.

Reply:
The `drec_del()` will remove the dentry of the path given in parameter of the function, if this last one exist.
The `drec_decref()` will decrement the dentry counter of the parent of the path given as parameter of the function.